src/xtgeo/grid3d/_egrid.py

```diff
@@ -277,8 +277,15 @@ def _check_xtgeo_compatible(self):
                 "Xtgeo does not currently support cylindrical coordinate systems"
             )
         if self.grid_head.numres < 1:
-            raise ValueError("EGrid file given with numres < 1")
-        if self.grid_head.numres != 1:
+            warnings.warn(
+                " EGrid file given with numres < 1, this is invalid, "
+                " assuming that the file contains exactly one reservoir,"
+                " which might fail. XTGeo prior to version 2.14 would"
+                " output grids with numres == 0. If the file was created"
+                " with an older version of XTGeo, please import with version 2.15"
+                " and re-export."
```

Comment:
Just a quick question, will this actually change `grid_head`? It won't just write the same number as has been read?

Reply:
I have to admit, I fail to see why a default `grid_head` init proves that `numres` is not overwritten with the old value*
 :P The other properties of `grid_head` are obviously updated? 
 
 * which to some extent sounds like a bug :P. I would expect handling of the `numres` on import have a check if it's below 1, and if so, set it to 1 and provide the warning where it's stated that `grid_head` set to 1 for compatibility reasons (or similar)